flocker/control/_model.py

```diff
@@ -376,7 +379,19 @@ def __invariant__(self):
                     return (False, '%r manifestation is not on node' % (app,))
         return (True, "")
 
+    def __new__(cls, **kwargs):
+        # PRecord does some crazy stuff, thus _precord_buckets; see
+        # PRecord.__new__.
+        if "uuid" not in kwargs and "_precord_buckets" not in kwargs:
+            warn("UUID is required, this is for backwards compat with existing"
+                 " tests only. If you see this in production code that's "
+                 "a bug.", DeprecationWarning, stacklevel=2)
+            kwargs["uuid"] = ip_to_uuid(kwargs["hostname"])
+        return PRecord.__new__(cls, **kwargs)
```

Comment:
Honestly I hope this code goes away ASAP, so maybe just worth spending the time in near future to remove the backwards compatibility code if e.g. pyrsistent changes.